File: app/src/main/java/com/neverlands/anlc/forms/ProfilesActivity.java
package com.neverlands.anlc.forms;

import android.app.Activity;
import android.os.Bundle;
import com.neverlands.anlc.auth.AuthRepository;
import com.neverlands.anlc.auth.ProfileManager;

import java.util.List;
import java.util.stream.Collectors;

public class ProfilesActivity extends Activity {
    private android.widget.LinearLayout layout;

        @Override
        protected void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            layout = new android.widget.LinearLayout(this);
            layout.setOrientation(android.widget.LinearLayout.VERTICAL);
            layout.setPadding(32, 32, 32, 32);
            setContentView(layout);
            AuthRepository.INSTANCE.init(getApplicationContext());
        }

        @Override
        protected void onResume() {
            super.onResume();
            layout.removeAllViews();

            // Принудительно обновляем список профилей
            List<ProfileManager.Profile> profiles = ProfileManager.INSTANCE.getProfiles();

            if (profiles.isEmpty()) {
                android.widget.TextView emptyText = new android.widget.TextView(this);
                emptyText.setText("Нет профилей. Создайте новый.");
                emptyText.setTextSize(18);
                emptyText.setPadding(0, 0, 0, 32);
                layout.addView(emptyText);
            } else {
                android.widget.TextView listLabel = new android.widget.TextView(this);
                listLabel.setText("Выберите профиль:");
                listLabel.setTextSize(16);
                listLabel.setPadding(0, 0, 0, 16);
                layout.addView(listLabel);

                android.widget.ListView listView = new android.widget.ListView(this);
                android.widget.ArrayAdapter<String> adapter = new android.widget.ArrayAdapter<>(this, android.R.layout.simple_list_item_1, profiles.stream().map(p -> p.getLogin()).collect(Collectors.toList()));
                listView.setAdapter(adapter);
                layout.addView(listView);

                // Кнопка входа
                android.widget.Button loginButton = new android.widget.Button(this);
                loginButton.setText("Войти в выбранный профиль");
                loginButton.setTextSize(16);
                layout.addView(loginButton);

                final int[] selectedIndex = {-1};
                listView.setOnItemClickListener((parent, view, position, id) -> {
                    selectedIndex[0] = position;
                });

                loginButton.setOnClickListener(v -> {
                    if (selectedIndex[0] >= 0 && selectedIndex[0] < profiles.size()) {
                        ProfileManager.Profile profile = profiles.get(selectedIndex[0]);
                        if (profile != null) {
                            android.widget.Toast.makeText(this, "Авторизация...", android.widget.Toast.LENGTH_SHORT).show();
                            AuthRepository.INSTANCE.authorizeAsync(profile, (result, body, error) -> {
                                handleAuthResult(result, body, error, profile);
                                return null;
                            });
                        } else {
                            android.widget.Toast.makeText(this, "Профиль не найден", android.widget.Toast.LENGTH_SHORT).show();
                        }
                    } else {
                        android.widget.Toast.makeText(this, "Выберите профиль", android.widget.Toast.LENGTH_SHORT).show();
                    }
                });
            }

            // Кнопка создания профиля всегда внизу
            android.widget.Button createButton = new android.widget.Button(this);
            createButton.setText("Создать профиль");
            createButton.setTextSize(16);
            layout.addView(createButton);

            createButton.setOnClickListener(v -> {
                android.content.Intent intent = new android.content.Intent(this, com.neverlands.anlc.forms.ProfileActivity.class);
                intent.putExtra("is_new", true);
                startActivity(intent);
            });
        }

    private void handleAuthResult(int result, String body, String error, ProfileManager.Profile profile) {
        if (result == 0) {
            android.widget.Toast.makeText(this, "Авторизация успешна", android.widget.Toast.LENGTH_SHORT).show();
            AuthRepository.INSTANCE.startSessionHeartbeat(profile);
            android.content.Intent intent = new android.content.Intent(this, com.neverlands.anlc.abforms.MainActivity.class);
            startActivity(intent);
            finish();
        } else {
            android.widget.Toast.makeText(this, "Ошибка авторизации: " + error, android.widget.Toast.LENGTH_LONG).show();
        }
    }
}

File: app/src/main/java/com/neverlands/anlc/ui/login/ProfilesActivity.kt
package com.neverlands.anlc.ui.login

import android.content.Intent
import android.os.Bundle
import android.widget.ArrayAdapter
import androidx.activity.viewModels
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.neverlands.anlc.R
import com.neverlands.anlc.data.local.ProfileManager
import com.neverlands.anlc.data.local.model.Profile
import com.neverlands.anlc.data.remote.AuthResult
import com.neverlands.anlc.databinding.ActivityProfilesBinding
import com.neverlands.anlc.forms.LogListActivity
import com.neverlands.anlc.ui.base.BaseActivity
import com.neverlands.anlc.ui.main.MainActivity

class ProfilesActivity : BaseActivity() {

    private val viewModel: ProfileViewModel by viewModels { ProfileViewModelFactory(application) }
    private lateinit var binding: ActivityProfilesBinding
    private var selectedProfile: Profile? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityProfilesBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setupViews()
        setupObservers()
    }

    override fun onResume() {
        super.onResume()
        // Load profiles when the activity resumes to ensure the list is up-to-date
        // This handles cases where a profile is added/edited/deleted in ProfileEditActivity
        // and we return to ProfilesActivity.
        ProfileManager.profiles.value?.let { currentProfiles ->
            val adapter = ArrayAdapter(
                this,
                android.R.layout.simple_list_item_single_choice,
                currentProfiles.map { it.userNick }
            )
            binding.profilesListView.adapter = adapter
            binding.profilesListView.choiceMode = android.widget.ListView.CHOICE_MODE_SINGLE
            binding.profilesListView.setOnItemClickListener { _, _, position, _ ->
                selectedProfile = currentProfiles[position]
            }
        }
    }

    private fun setupViews() {
        binding.loginButton.setOnClickListener {
            selectedProfile?.let { profile ->
                PasswordInputDialog { password ->
                    viewModel.authorize(profile, password)
                }.show(supportFragmentManager, "password_dialog")
            } ?: showToast("Пожалуйста, выберите профиль")
        }

        binding.createButton.setOnClickListener {
            val intent = Intent(this, ProfileEditActivity::class.java)
            intent.putExtra("is_new_profile", true)
            startActivity(intent)
        }

        binding.editButton.setOnClickListener {
            selectedProfile?.let { profile ->
                val intent = Intent(this, ProfileEditActivity::class.java)
                intent.putExtra("is_new_profile", false)
                intent.putExtra("profile_to_edit", profile)
                startActivity(intent)
            } ?: showToast("Пожалуйста, выберите профиль для редактирования")
        }

        binding.deleteButton.setOnClickListener {
            selectedProfile?.let { profile ->
                viewModel.removeProfile(profile)
                selectedProfile = null
            } ?: showToast("Пожалуйста, выберите профиль для удаления")
        }

        binding.logsButton.setOnClickListener {
            val intent = Intent(this, LogListActivity::class.java)
            startActivity(intent)
        }
    }

    private fun setupObservers() {
        viewModel.profiles.observe(this) { profiles ->
            val adapter = ArrayAdapter(
                this,
                android.R.layout.simple_list_item_single_choice,
                profiles.map { it.userNick }
            )
            binding.profilesListView.adapter = adapter
            binding.profilesListView.choiceMode = android.widget.ListView.CHOICE_MODE_SINGLE
            binding.profilesListView.setOnItemClickListener { _, _, position, _ ->
                selectedProfile = profiles[position]
            }
        }

        viewModel.authResult.observe(this) { result ->
            when (result) {
                is AuthResult.Success -> {
                    selectedProfile?.let { profile ->
                        ProfileManager.setCurrentProfile(profile)
                        val intent = Intent(this, MainActivity::class.java).apply {
                            if (result.useHolder) {
                                putExtra("use_holder", true)
                            } else {
                                putExtra("html_content", result.htmlContent)
                            }
                        }
                        startActivity(intent)
                        finish()
                    }
                }
                is AuthResult.Failure -> showToast("Ошибка: ${result.error}")
            }
        }
    }
}

class ProfileViewModelFactory(private val application: android.app.Application) : ViewModelProvider.Factory {
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(ProfileViewModel::class.java)) {
            @Suppress("UNCHECKED_CAST")
            return ProfileViewModel(application) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}